mdata/store_cassandra.go

```diff
@@ -77,9 +78,37 @@ type cassandraStore struct {
 	session     *gocql.Session
 	writeQueues []chan *ChunkWriteRequest
 	readQueue   chan *ChunkReadRequest
+	ttlTables   map[uint32]string
 }
 
-func NewCassandraStore(addrs, keyspace, consistency, CaPath, Username, Password, hostSelectionPolicy string, timeout, readers, writers, readqsize, writeqsize, retries, protoVer int, ssl, auth, hostVerification bool) (*cassandraStore, error) {
+func (c *cassandraStore) ensureTtlTables(ttls []uint32, keyspace string) error {
+	for _, ttl := range ttls {
+		// calculate table name and window
+		var window int = 1
+		for tmp := ttl / (24 * 60 * 60); tmp > 1; tmp = tmp / 2 {
+			window *= 2
+		}
+		table_name := fmt.Sprintf("metric_%d", window)
+
+		// table isn't know to exist, ensure it exists
+		err := c.session.Query(fmt.Sprintf(table_schema, keyspace, table_name, window)).Exec()
```

Comment:
> In your comment you mean this, right? window_size := (ttl/20) +1

no i mean (window/20 + 1)

ttl range (days) | table name | window_size
-----------|----------------|----------------
0-1 | metrics_1 | 1day
2-3 | metrics_2 | 1day
4-7 | metrics_4 | 1day
8-15| metrics_8 | 1day
16-31| metrics_16 | 1day
32-63 | metrics_32 | 2days
64-127| metrics_64 | 4days
128-256| metrics_128 | 7days


> why does it have to be a loop? can't you make it a formula? 

I really like this approach of ^2 buckets.  But it does need better explaining. Including the above table in a comment would be good.